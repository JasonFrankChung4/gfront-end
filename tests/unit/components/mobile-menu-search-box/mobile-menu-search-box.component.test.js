import React from 'react'
import { shallow } from 'enzyme'

import MobileMenuSearchBox from 'components/mobile-menu-search-box/mobile-menu-search-box.component'

const props = {
  onSubmit: jest.fn(),
}

let wrapper
let instance
describe('components - MobileMenuSearchBox', () => {
  beforeEach(() => {
    wrapper = shallow(<MobileMenuSearchBox {...props} />)
    instance = wrapper.instance()
  })

  describe('handleChange', () => {
    beforeEach(() => {
      instance.handleChange({ target: { value: 'test-value' } })
    })

    it('should set the value', () => {
      expect(wrapper.state('value')).toBe('test-value')
    })
  })

  describe('handleSubmit', () => {
    beforeEach(() => {
      wrapper.setState({ value: 'test-value' })
      instance.handleSubmit({ preventDefault: jest.fn() })
    })

    it('should call `onSubmit` with the value', () => {
      expect(props.onSubmit).toBeCalledWith('test-value')
    })

    it('should clear the value', () => {
      expect(wrapper.state('value')).toBe('')
    })
  })

  describe('render', () => {
    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })
})