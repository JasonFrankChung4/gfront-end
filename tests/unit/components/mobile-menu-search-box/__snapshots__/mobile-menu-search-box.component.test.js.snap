// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components - MobileMenuSearchBox render should render correctly 1`] = `
<form
  className="search-form"
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Search Projects..."
    type="search"
    value=""
  />
  <button
    className="go"
  >
    Go
  </button>
</form>
`
